### PR TITLE
chore: migrate zklogin and multisig tests to simtest

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/unit_tests/transaction_tests.rs [hunk 1]
```diff
@@ -453,15 +451,15 @@ async fn test_zklogin_transfer_with_bad_ephemeral_sig() {
     .await;
 }
 
-fn zklogin_key_pair_and_inputs() -> Vec<(Ed25519KeyPair, ZkLoginInputs)> {
-    let key1 = Ed25519KeyPair::generate(&mut StdRng::from_seed([1; 32]));
-    let key2 = Ed25519KeyPair::generate(&mut StdRng::from_seed([2; 32]));
+// fn zklogin_key_pair_and_inputs() -> Vec<(Ed25519KeyPair, ZkLoginInputs)> {
```
Comment (lxfind) [inline crates/sui-core/src/unit_tests/transaction_tests.rs:454]: Is this still WIP?

Comment (joyqvq) [reply crates/sui-core/src/unit_tests/transaction_tests.rs]: its ready for review! just pushed some lints and fixes

File: crates/test-cluster/src/lib.rs [hunk 2]
```diff
@@ -1058,3 +1058,24 @@ impl Default for TestClusterBuilder {
         Self::new()
     }
 }
+
+pub async fn fund_address_and_return_gas(
+    context: &WalletContext,
+    rgp: u64,
+    funding_address: SuiAddress,
+) -> ObjectRef {
+    // fund the multisig address.
```
Comment (lxfind) [inline crates/test-cluster/src/lib.rs:1067]: Does the address have to be multisig?

File: crates/test-cluster/src/lib.rs [hunk 3]
```diff
@@ -1058,3 +1058,24 @@ impl Default for TestClusterBuilder {
         Self::new()
     }
 }
+
+pub async fn fund_address_and_return_gas(
+    context: &WalletContext,
```
Comment (lxfind) [inline crates/test-cluster/src/lib.rs:1063]: Could this be a method of TestCluster? If so, it's generally preferred for discoverability. Global functions are really hard to discover.

File: crates/test-cluster/src/lib.rs [hunk 4]
```diff
@@ -1058,3 +1058,24 @@ impl Default for TestClusterBuilder {
...
+    rgp: u64,
+    funding_address: SuiAddress,
+) -> ObjectRef {
+    // fund the multisig address.
+    let (sender, gas) = context.get_one_gas_object().await.unwrap().unwrap();
+    let tx = context.sign_transaction(
+        &TestTransactionBuilder::new(sender, gas, rgp)
+            .transfer_sui(Some(20000000000), funding_address)
```
Comment (lxfind) [inline crates/test-cluster/src/lib.rs:1071]: Consider making the amount an argument

File: crates/test-cluster/src/lib.rs [hunk 5]
```diff
@@ -1058,3 +1058,24 @@ impl Default for TestClusterBuilder {
         Self::new()
     }
 }
+
+pub async fn fund_address_and_return_gas(
```
Comment (lxfind) [inline crates/test-cluster/src/lib.rs:1062]: Some doc comment would be good

File: crates/sui-sdk/src/wallet_context.rs [hunk 6]
```diff
@@ -277,7 +277,15 @@ impl WalletContext {
         let gas_price = client.governance_api().get_reference_gas_price().await?;
         Ok(gas_price)
     }
-
+    pub async fn get_current_epoch(&self) -> Result<u64, anyhow::Error> {
```
Comment (lxfind) [inline crates/sui-sdk/src/wallet_context.rs:280]: Is this actually used?

**Review by lxfind — APPROVED** [review]

**Pull request merged**